sound/soc/sof/intel/hda-bus.c

```diff
@@ -19,11 +20,33 @@
 #define sof_hda_ext_ops	NULL
 #endif
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_LINK)
+static void sof_hda_bus_link_power(struct hdac_device *codec, bool enable)
+{
+	struct hdac_bus *bus = codec->bus;
+	bool oldstate = test_bit(codec->addr, &bus->codec_powered);
+
+	snd_hdac_ext_bus_link_power(codec, enable);
+
+	if (enable == oldstate)
+		return;
+
+#if IS_ENABLED(CONFIG_SND_HDA_CODEC_HDMI) || IS_ENABLED(CONFIG_SND_SOC_HDAC_HDMI)
+	/*
+	 * when display codec is suspended, we can release display
+	 * power
+	 */
+	if (HDA_IDISP_CODEC(codec->addr) && !enable)
+		snd_hdac_display_power(bus, HDA_CODEC_IDX_CONTROLLER, false);
```

Comment:
dont we need to re-enable it when link_power is turned back on?

Reply:
@ranj063 hda_codec.c:hda_codec_resume() will power up the display. we only need to power up the display hw when we do a reinit with hda_dsp_ctrl_init_chip() and there's no guarantee codec driver is active.